### PR TITLE
[clang][OpenMP] Add 'allocator' modifier for 'allocate' clause.

### PR DESCRIPTION
**Pull request by ddpagan**
The 'allocator' modifier is now accepted in the 'allocate' clause. Added LIT tests covering codegen, PCH, template handling, and serialization for 'allocator' modifier.

Added support for allocator-modifier to release notes.

Testing
- New allocate modifier LIT tests.
- OpenMP LIT tests.
- check-all
- relevant sollve_vv test cases tests/5.2/scope/test_scope_allocate_construct.c

### CODE REVIEW

**Comment by ddpagan** [comment]
NOTE: OpenMP Support doc will be updated as well now that the PR # is available.

File: clang/lib/AST/OpenMPClause.cpp [hunk 1]
```diff
@@ -2242,9 +2247,16 @@ void OMPClausePrinter::VisitOMPAllocateClause(OMPAllocateClause *Node) {
...
   if (Expr *Allocator = Node->getAllocator()) {
     OS << "(";
-    Allocator->printPretty(OS, nullptr, Policy, 0);
+    if (Modifier == OMPC_ALLOCATE_allocator) {
+      OS << getOpenMPSimpleClauseTypeName(Node->getClauseKind(), Modifier);
+      OS << "(";
+      Allocator->printPretty(OS, nullptr, Policy, 0);
+      OS << ")";
+    } else
+      Allocator->printPretty(OS, nullptr, Policy, 0);
```
Comment (alexey-bataev) [inline clang/lib/AST/OpenMPClause.cpp:2259]:
```suggestion
    } else {
      Allocator->printPretty(OS, nullptr, Policy, 0);
    }
```

File: clang/lib/Parse/ParseOpenMP.cpp [hunk 2]
```diff
@@ -4539,6 +4539,8 @@ bool Parser::ParseOpenMPVarList(OpenMPDirectiveKind DKind,
   bool NeedRParenForLinear = false;
   BalancedDelimiterTracker LinearT(*this, tok::l_paren,
                                    tok::annot_pragma_openmp_end);
+  BalancedDelimiterTracker AllocateT(*this, tok::l_paren,
+                                     tok::annot_pragma_openmp_end);
```
Comment (alexey-bataev) [inline clang/lib/Parse/ParseOpenMP.cpp:4543]: Move it exactly where it used (lines 4807-4819)

File: clang/lib/AST/OpenMPClause.cpp (hunk 1 above)
Comment (ddpagan) [reply clang/lib/AST/OpenMPClause.cpp:2259]: Fixed as suggested.

File: clang/lib/Parse/ParseOpenMP.cpp (hunk 2 above)
Comment (ddpagan) [reply clang/lib/Parse/ParseOpenMP.cpp:4543]: Fixed as suggested.

**Review by alexey-bataev — APPROVED** [review]
LG with a nit. Update OpenMPSupport.rst

**Comment by ddpagan** [comment]
@alexey-bataev - I'll make that change in the next PR. Thanks for the review.

**Pull request merged**